src/guards.js

```diff
@@ -0,0 +1,30 @@
+export function isLegacy(functionMap) {
+  return function(key) {
+    return key in functionMap
+      && functionMap[key] !== undefined
+      && (functionMap[key].timing === undefined);
+  };
+}
+
+export function isImmediate(functionMap) {
+  return function(key) {
+    return key in functionMap
+      && (functionMap[key] === undefined || functionMap[key].timing === 'immediate');
```

Comment:
The `undefined` condition is not applicable anymore for `isImmediate` since `isLegacy` covers that now, right?
```suggestion
      && functionMap[key] !== undefined
      && functionMap[key].timing === 'immediate';
```